File: C/Grace.c
#include <fcntl.h>
#include <stdio.h>
#define FT(x) int main() { FILE* f = fopen("Grace_kid.c", "w"); if (f == 0) return 1; fputs("#include <fcntl.h>\n#include <stdio.h>\n#define FT(x) "x"\n#define STR(x) #x\n#define EXPAND(x) STR(x)\n\n/*\n This program will print its own source when run.\n*/\n\nFT(EXPAND(FT(x)))\n", f); fclose(f); return 0; }
#define STR(x) #x
#define EXPAND(x) STR(x)

/*
 This program will print its own source when run.
*/

FT(EXPAND(FT(x)))

File: Python/Grace.py
import sys

# This program will print its own source when run.

codeStr = "import sys\n\n# This program will print its own source when run.\n\ncodeStr = \"{0}\"\n\ndef main():\n    try:\n        with open(\"Grace_kid.py\", \"w\") as file:\n            file.write(str.format(str.replace('\\\\', '\\\\\\\\').replace('\"', '\\\\\"').replace('\\n', '\\\\n')))\n    except:\n        print(\"Error writing file!\")\n        sys.exit(1)\n\nif __name__ == \"__main__\":\n    main()\n"

def main():
    try:
        with open("Grace_kid.py", "w") as file:
            file.write(str.format(str.replace('\\', '\\\\').replace('"', '\\"').replace('\n', '\\n')))
    except:
        print("Error writing file!")
        sys.exit(1)

if __name__ == "__main__":
    main()
